wl-sim/main/WL_Flash.cpp

```diff
@@ -0,0 +1,292 @@
+#include <cmath>
+#include "esp_log.h"
+#include "esp_err.h"
+
+#include "wl_sim.h"
+
+static const char *TAG = "WL_Flash";
+
+#define SECTOR_ERASE_ENDURANCE 100000
+
+size_t access_count = 0;
```

Comment:
Nitpicking:
* if the global variables are not accessed from other modules, it is a good practise make them static (so they are limited to the scope of the file where defined)
* once in CPP, it would be useful to define a namespace class as the global vars holder 
* once static, they should be prefixed with 's_'. (ESP IDF rules, probably new to you)

Reply:
first point addressed, others intentionally 'ignored'; if they remain something that will be desired or required in the future, I'd conform